microbench/src/main/java/alluxio/fsmaster/FileSystemMasterBench.java

```diff
@@ -0,0 +1,156 @@
+import org.openjdk.jmh.results.format.ResultFormatType;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.CommandLineOptionException;
+import org.openjdk.jmh.runner.options.CommandLineOptions;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Benchmarks for {@link alluxio.master.file.FileSystemMaster}.
+ */
+@Fork(value = 1, jvmArgsPrepend = "-server")
+@Warmup(iterations = 2, time = 3, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 6, time = 3, timeUnit = TimeUnit.SECONDS)
+@BenchmarkMode(Mode.Throughput)
+public class FileSystemMasterBench {
+
+  @State(Scope.Thread)
+  public static class ThreadState extends BaseThreadState {
+    @Setup(Level.Trial)
+    public void setup() {
+      AuthenticatedClientUser.set("test");
+    }
+  }
+
+  @State(Scope.Benchmark)
+  public static class FileSystem extends BaseFileStructure {
+    FileSystemMasterBase mBase = new FileSystemMasterBase();
+
+    @Setup(Level.Trial)
+    public void setup() throws Exception {
+      mBase.init();
+      mBase.createPathDepths(mDepth);
+      for (int d = 0; d < mDepth + 1; d++) {
+        for (int i = 0; i < mFileCount; i++) {
+          mBase.createFile(d, i);
+        }
+      }
+    }
+
+    @TearDown
+    public void tearDown() throws Exception {
+      mBase.tearDown();
+    }
+  }
+
+  private <T> ServerCallStreamObserver<T> createStreamObserver(Blackhole bh) {
+    return new ServerCallStreamObserver<T>() {
+      @Override
+      public boolean isCancelled() {
+        return false;
+      }
+
+      @Override
+      public void setOnCancelHandler(Runnable onCancelHandler) {
+        bh.consume(onCancelHandler);
+      }
+
+      @Override
+      public void setCompression(String compression) {
+        bh.consume(compression);
+      }
+
+      @Override
+      public boolean isReady() {
+        return true;
+      }
+
+      @Override
+      public void setOnReadyHandler(Runnable onReadyHandler) {
+        bh.consume(onReadyHandler);
+      }
+
+      @Override
+      public void disableAutoInboundFlowControl() {
+        bh.consume(new Object());
+      }
+
+      @Override
+      public void request(int count) {
+        bh.consume(count);
+      }
+
+      @Override
+      public void setMessageCompression(boolean enable) {
+        bh.consume(enable);
+      }
+
+      @Override
+      public void onNext(T value) {
+        bh.consume(value);
+      }
+
+      @Override
+      public void onError(Throwable t) {
+        throw new RuntimeException(t);
+      }
+
+      @Override
+      public void onCompleted() {
+        bh.consume(new Object());
+      }
+    };
+  }
+
+  @Benchmark
+  public void getStatusBench(FileSystem fs, ThreadState ts, Blackhole bh) {
+    ServerCallStreamObserver<GetStatusPResponse> so = createStreamObserver(bh);
```

Comment:
You probably don't need to create this object on every call. You can probably just create a single on the the Base class then reuse it for all calls.